### PR TITLE
Allow index argument in for expressions.

### LINKED ISSUE #418: Add index to for loop when using arrays
If you need the index to perform actions on an array, it seems like you have to use `Array.mapWithIndex`. This means that you need to create a computed property when you need to use the index in the render function.
It would be nice to be able to get the index in a normal for loop, then you could simply get rid of the computed property completely.

Example: https://sandbox.mint-lang.com/sandboxes/bmHCM6s4Kn8RgA

### PR DESCRIPTION
**Pull request by gdotdesign**
This PR implements the `index` as the last argument for `for` expressions:

it's the second argument for `Set(a)` and `Array(a)`:

```
for (item, index of ["Item1", "Item2"]) {
  index
}
```

and the third argument for `Map(key, value)`:

```
for (key, value, index of Map.empty()) {
  key
} 
```

Resolves #418

### CODE REVIEW

File: src/compilers/for_expression.cr [hunk 1]
```diff
@@ -22,18 +38,24 @@ module Mint
           JS
         end
 
+      index =
+        if index_arg
+          "const #{js.variable_of(index_arg.not_nil!)} = _i"
```
Comment (Sija) [inline src/compilers/for_expression.cr:43]:
```suggestion
          "const #{js.variable_of(index_arg)} = _i"
```

File: src/type_checkers/for_expression.cr [hunk 2]
```diff
@@ -25,18 +25,22 @@ module Mint
 
       raise ForMapArgumentsMismatch, {
         "node" => node,
-      } if is_map && node.arguments.size != 2
+      } if is_map && ![2, 3].includes?(node.arguments.size)
```
Comment (Sija) [inline src/type_checkers/for_expression.cr:28]:
```suggestion
      } if is_map && !node.arguments.size.in?(2, 3)
```

File: src/type_checkers/for_expression.cr [hunk 3]
```diff
@@ -25,18 +25,22 @@ module Mint
...
 
       raise ForArrayOrSetArgumentsMismatch, {
         "node" => node,
-      } if is_array_or_set && node.arguments.size != 1
+      } if is_array_or_set && ![1, 2].includes?(node.arguments.size)
```
Comment (Sija) [inline src/type_checkers/for_expression.cr:32]:
```suggestion
      } if is_array_or_set && !node.arguments.size.in?(2, 3)
```

File: src/compilers/for_expression.cr (hunk 1 above)
Comment (gdotdesign) [reply src/compilers/for_expression.cr:43]: I thought there is no overload for `variable_of(variable : Nil)`

Comment (Sija) [reply src/compilers/for_expression.cr]: `index_arg` can't be `Nil` at this point.

Comment (gdotdesign) [reply src/compilers/for_expression.cr]: Yep I figured :facepalm:

**Review by Sija — APPROVED** [review]
LGTM

**Pull request merged**